modules/node/src/admin/admin.service.ts

```diff
@@ -1,21 +1,52 @@
-import { StateChannelJSON } from "@connext/types";
+import { ConnextNodeStorePrefix, StateChannelJSON } from "@connext/types";
 import { Injectable } from "@nestjs/common";
 
+import { CFCoreRecordRepository } from "../cfCore/cfCore.repository";
 import { CFCoreService } from "../cfCore/cfCore.service";
 import { Channel } from "../channel/channel.entity";
 import { ChannelService } from "../channel/channel.service";
+import { ConfigService } from "../config/config.service";
 import { LinkedTransfer } from "../transfer/transfer.entity";
 import { TransferService } from "../transfer/transfer.service";
-import { CLogger } from "../util";
+import { CLogger, getCreate2MultisigAddress } from "../util";
 
 const logger = new CLogger("AdminService");
 
+const HISTORICAL_PROXY_FACTORY_ADDRESSES = {
+  1: [
+    "0x90Bf287B6870A99E32130CED0Da8b02302a8a4dE",
+    "0xA16d9511C743d6D6177A65892DC2Eafd417BfD7A",
+    "0xc756Bf6A685573C6879D4363401940f02B4E27a1",
+    "0x6CF0c4Ab3F1e66913c0983DC0bb1202d958ABb8f",
+    "0x711C655e08aaA9081e0BDc431920507CCD96b7a0",
+    "0xF9015aA98BeBaE3e43945c48dc3fB6c0a5281986",
+  ],
+  4: [
+    "0x49f2eCa045B2372C334B6CcBB9232C48f9acA097",
+    "0x8A49B435cc3D2176B67e0D26170387EeDf135669",
+    "0xD891F41c4ba30b1FF4f604e30F64ae387DD85b4F",
+    "0x6CF0c4Ab3F1e66913c0983DC0bb1202d958ABb8f",
+    "0xc8d7Cf5638dfa5b79A070c5aC983716575bEF4B0",
+    "0xc40E9B210363163a143f454fa505ACeAA28Cd475",
+    "0x8eb543b35DE94B0E636402C7cA32947b22853eDF",
+    "0xc8d7Cf5638dfa5b79A070c5aC983716575bEF4B0",
+    "0x8eb543b35DE94B0E636402C7cA32947b22853eDF",
+  ],
+};
```

Comment:
noice, wondering if there's a good way to keep this up to date in an automated way?